src/View/AbstractTreeView.vala

```diff
@@ -349,6 +352,49 @@ namespace FM {
     }
 
     protected class TreeView : Gtk.TreeView {
+        private Marlin.ZoomLevel _zoom_level = Marlin.ZoomLevel.INVALID;
+        public Marlin.ZoomLevel zoom_level {
+            set {
+                if (_zoom_level == value || !get_realized ()) {
+                    return;
+                } else {
+                    _zoom_level = value;
+                }
+
+                /* Make rows closer together at small icon sizes.  There is no property for this so we have to use
+                 * css.
+                 */
+                int vsep = 5;
+                switch (value) {
+                    case Marlin.ZoomLevel.SMALLEST:
+                        vsep = 0;
+                        break;
+                    case Marlin.ZoomLevel.SMALLER:
+                        vsep = 3;
+                        break;
+                    default:
+                        break;
+                }
+
+                try {
+                    view_css.load_from_data ("treeview { -GtkTreeView-vertical-separator: %i px; }".printf (vsep));
+                    get_style_context ().add_provider (view_css, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+                } catch (Error e) {
+                    warning ("Could not style FM.TreeView");
+                }
+            }
```

Comment:
I would really prefer not to do this. If the padding is too large for tree view rows, we should address that in the stylesheet and not special case Files styling

Reply:
I agree in principle - this is more by way of a demonstration.  The padding only changes for small icons - can you accomplish that in the stylesheet? Would you add two new classes for the treeview with different padding?

Reply:
I'm not sure we really need to have two different paddings. Maybe we can open an issue in the stylesheet to reconsider the default and see where we wind up?

Reply:
The row padding is reduced at small icon size to get the maximum benefit in terms of compactness a well as avoiding an unbalanced appearance (space between rows too large compared with the row content).  There are in fact three different paddings used in this PR (0, 3 and 5 px).  But I agree it is ultimately up to the UX team to arrive at the appropriate compromise. I'll open an issue in stylesheet.  Is this PR mergeable if I remove the CSS or shall I put it on the backburner?